plugins/janus_streaming.c

```diff
@@ -5173,6 +5173,20 @@ static int janus_streaming_rtsp_connect_to_server(janus_streaming_mountpoint *mp
 			curl_easy_cleanup(curl);
 			g_free(curldata->buffer);
 			g_free(curldata);
+			close(video_fds.fd);
+			close(video_fds.rtcp_fd);
+			close(audio_fds.fd);
+			close(audio_fds.rtcp_fd);
+			return -5;
```

Comment:
That's right. Updated to close only when fd != -1.
However I feel like these repetitive clean-up codes are somewhat ugly.
Maybe you could re-implement it in a more cleaner way instead of directly merging this pull-req as my contribution. I'm fine with it.

Reply:
I don't mind them that much, it's just 4 occurrences limited to a single specific function. It could be done with a `goto error:` kind of thing, or with a dedicated function, but I think it's overkill. I'll merge as it is, and we can worry about prettifying in the future :smile: